### ISSUE TITLE
mapEnv command fails with "TypeError: Cannot read property 'split' of undefined"

### ISSUE THREAD

**Issue by elson** [prescottprue/firebase-ci#46]
**Do you want to request a *feature* or report a *bug*?**  

bug

**What is the current behavior?**

Running `mapEnv` locally using the node_module bin fails with a type error.

This seems to be because `runCommand` expects an object, but `mapEnv` passes it a string.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar.**

Example .firebaserc file:

```json
{
  "projects": {
    "default": "my-project",
    "master": "my-project"
  },
  "ci": {
    "mapEnv": {
      "SOME_VAR": "some.var"
    }
  }
}
```

Call to mapEnv:

```bash
./node_modules/firebase-ci/bin/firebase-ci mapEnv
ℹ Mapping Environment to Firebase Functions...
✖ Error: Error setting Firebase functions config variables from variables CI environment (mapEnv):

 TypeError: Cannot read property 'split' of undefined
    at Promise (/Users/elson/Projects/trello-sync/node_modules/firebase-ci/lib/utils/commands.js:41:80)
    at new Promise (<anonymous>)
    at runCommand (/Users/elson/Projects/trello-sync/node_modules/firebase-ci/lib/utils/commands.js:40:10)
    at /Users/elson/Projects/trello-sync/node_modules/firebase-ci/lib/actions/mapEnv.js:95:64
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/elson/Projects/trello-sync/node_modules/firebase-ci/lib/actions/mapEnv.js:36:103)
    at _next (/Users/elson/Projects/trello-sync/node_modules/firebase-ci/lib/actions/mapEnv.js:38:194)
    at /Users/elson/Projects/trello-sync/node_modules/firebase-ci/lib/actions/mapEnv.js:38:364
    at new Promise (<anonymous>)
    at /Users/elson/Projects/trello-sync/node_modules/firebase-ci/lib/actions/mapEnv.js:38:97
```

**What is the expected behavior?**

Firebase functions variables should be set as expected with no error thrown.

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**

firebase-ci: 0.6.0
Node: 11.14.0
OSX: 10.14.5

**Workarounds**

I have managed to make this run as expected by editing line 92 of mapEnv.js, so a suitable object is passed to `runCommand`:

```javascript
const setConfigCommand = {
  command: createConfigSetString(mapEnvSettings)
};
```

However, on doing this I noted a separate issue with the resulting firebase function variable strings containing escaped `"` characters, e.g.:

```bash
firebase functions:config:get
{
    "some": {
        "var": "\"my value\""
    }
}
```

Passing the `shell: true` option to `spawn` in commands.js fixes this issue:

```javascript
const child = spawn((0, _isArray2.default)(command) ? command[0] : command.split(' ')[0], args || (0, _compact2.default)((0, _drop2.default)(command.split(' '))), {
      env: process.env,
      shell: true
    });
```

**Comment by prescottprue** [comment]
Thanks for reaching out!

Working on a fix to go into v0.7.0 - I included `shell: true` as well as fixing how the command is passed. While working on it I also simplified the `runCommand` util which contained the original error you were reporting.

**Comment by elson** [comment]
Awesome, thanks Scott!

One other thing, I noticed that in the split undefined scenario deployments
would still succeed, so perhaps the error wasn't being propagated properly
to cause the deploy call to fail?

I'll try to work on a reproducible case later.

Thanks again for your help.

Steve

**Comment by prescottprue** [comment]
Fix in [v0.7.0](https://github.com/prescottprue/firebase-ci/releases/tag/v0.7.0). Reach out if things are still not working as expected
